pandas/io/parsers.py

```diff
@@ -2153,8 +2155,12 @@ def _try_convert_dates(parser, colspec, data_dict, columns):
 
     new_name = '_'.join([str(x) for x in colnames])
     to_parse = [data_dict[c] for c in colnames if c in data_dict]
-
     new_col = parser(*to_parse)
+    if hasattr(new_col, 'dtype') and any(r in str(new_col.dtype) for r in
```

Comment:
What? Oh, man. That was the first fix I came up with, scrubbed in favor of your suggestion. No idea how it made its way in.

the two first commits in the PR, b6d07b4 and 7d6f7c4, should be reverted. How do you propose I do this?